tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/ProtoSeqOps.scala

```diff
@@ -0,0 +1,268 @@
+package at.forsyte.apalache.tla.bmcmt.rules.aux
+
+import at.forsyte.apalache.tla.bmcmt.types.{CellT, UnknownT}
+import at.forsyte.apalache.tla.bmcmt.{Arena, ArenaCell, SymbState, SymbStateRewriter}
+import at.forsyte.apalache.tla.lir.TypedPredefs._
+import at.forsyte.apalache.tla.lir.convenience.tla
+import at.forsyte.apalache.tla.lir._
+
+/**
+ * Proto sequences that contain the absolute minimum for implementing TLA+ sequences. A proto sequence is essentially an
+ * array of fixed size (which we call 'capacity'). A sequence is a pair that consists of the length (an integer cell)
```

Comment:
I think analogies make a bad job here. The most important properties of proto sequences are:

 - a proto sequence has a fixed length (which I call `capacity` to avoid confusion with the length of a sequence), and
 - the elements of a proto sequence are ordered by indices in the range `1..capacity`.

Following Lamport, we could even say that proto sequences are actually tuples that have `capacity` elements. When we say "tuples" people bring other analogies though, so it does not help much.

Right, a sequence has the length, which may be not known in advance. This is the most important difference.